p2p/protocol/identify/id.go

```diff
@@ -113,27 +123,40 @@ func NewIDService(ctx context.Context, h host.Host, opts ...Option) *IDService {
 
 	// handle local protocol handler updates, and push deltas to peers.
 	var err error
-	s.subscription, err = h.EventBus().Subscribe(&event.EvtLocalProtocolsUpdated{}, eventbus.BufSize(128))
+	s.subscriptions.localProtocolsUpdated, err = h.EventBus().Subscribe(&event.EvtLocalProtocolsUpdated{}, eventbus.BufSize(128))
 	if err != nil {
 		log.Warningf("identify service not subscribed to local protocol handlers updates; err: %s", err)
 	} else {
-		go s.handleEvents()
+		go s.handleEvents(s.subscriptions.localProtocolsUpdated, s.handleProtosChanged)
+	}
+	s.subscriptions.localPeerRecordUpdated, err = h.EventBus().Subscribe(&event.EvtLocalPeerRecordUpdated{}, eventbus.BufSize(128))
+	if err != nil {
+		log.Warnf("identify service not subscribed to local routing state changes; err: %s", err)
+	} else {
+		go s.handleEvents(s.subscriptions.localPeerRecordUpdated, s.handlePeerRecordUpdated)
 	}
 
 	s.emitters.evtPeerProtocolsUpdated, err = h.EventBus().Emitter(&event.EvtPeerProtocolsUpdated{})
 	if err != nil {
 		log.Warningf("identify service not emitting peer protocol updates; err: %s", err)
 	}
 
+	hostKey := h.Peerstore().PrivKey(h.ID())
+	s.peerRecordManager, err = NewPeerRecordManager(ctx, h.EventBus(), hostKey, h.Addrs())
```

Comment:
Instead of passing the addrs, shouldn't we trigger an addrs change event up-front?

Reply:
Non-blocking: It also seems odd that we're constructing this inside the identify service (a specific protocol for exchanging peer information). I'd expect this to live at the top level inside the host (or possibly live inside the peerstore?).

Bug, given that it does live here and it's _private_, it shouldn't be hard to extract it (especially when we eventually get a dependency injection system.

Reply:
The peerRecordManager has now been moved to the Host.